app/models/miq_cockpit_ws_worker/authenticator.rb

```diff
@@ -0,0 +1,67 @@
+class MiqCockpitWsWorker::Authenticator
+  def authenticate_for_host(token, host)
```

Comment:
Having it return a hash makes dealing with drb better. So unless that changes I think keeping a hash is better. I don't think this belongs on the host model as it applies to lots of things that are not hosts.

It looks at key_pairs from vms and ssh_auth from container deployments. I could not find any way that a vm could have a password based credential stored. But even if that is not the case, I think we can expand this later to find more credentials as it becomes clear what is needed.